lib/OpenQA/CacheService.pm

```diff
@@ -30,7 +31,28 @@ sub startup {
     $self->defaults(appname => 'openQA Cache Service');
     # Provide help to users early to prevent failing later on
     # misconfigurations
-    return if $ENV{MOJO_HELP};
+    return $self->{_app_return_code} = 0 if $ENV{MOJO_HELP};
```

Comment:
I don't see any reason for this to be a an instance property. Couldn't it just be a variable in this scope?

Reply:
Just saw the test, then maybe store it in `$self->defaults->{app_exit_code}` or so, or make it an attribute of the app (`$app->exit_code`).

Reply:
If it's important enough for a test it's important enough for a public API. 😉

Reply:
If you really care I would actually create a sub for returning `_app_return_code` because too me it doesn't make any sense if the value is set from somewhere else. (So it should basically be a read-only property.)

Reply:
Not about this particular case. But i do think we have too many cases of private functions/instance data being used in tests. The underscore prefix for functions and instance data in general is just meant to be private in Perl. Makes it much easier to keep track of what might be a breaking change and what shouldn't be.

Reply:
In my opinion it is ok to use private functions in tests because the worst thing which can happen is that the tests fail.

Reply:
We'll probably have to agree to disagree on that. 😁 I strongly believe that a well designed API is a testable API.